src/features/encoders/wp2/client/index.tsx

```diff
@@ -18,93 +23,286 @@ interface Props {
 }
 
 interface State {
+  options: EncodeOptions;
+  effort: number;
+  quality: number;
+  alphaQuality: number;
+  passes: number;
+  sns: number;
+  uvMode: number;
+  lossless: boolean;
+  slightLoss: number;
+  colorSpace: number;
+  errorDiffusion: number;
+  useRandomMatrix: boolean;
   showAdvanced: boolean;
+  separateAlpha: boolean;
 }
 
 export class Options extends Component<Props, State> {
+  static getDerivedStateFromProps(
+    props: Props,
+    state: State,
+  ): Partial<State> | null {
+    if (state.options && shallowEqual(state.options, props.options)) {
+      return null;
+    }
+
+    const { options } = props;
+
+    const modifyState: Partial<State> = {
+      options,
+      effort: options.speed,
+      alphaQuality: options.alpha_quality,
+      passes: options.pass,
+      sns: options.sns,
+      uvMode: options.uv_mode,
+      colorSpace: options.csp_type,
+      errorDiffusion: options.error_diffusion,
+      useRandomMatrix: options.use_random_matrix,
+      separateAlpha: options.quality !== options.alpha_quality,
+    };
+
+    // If quality is > 95, it's lossless with slight loss
+    if (options.quality <= 95) {
+      modifyState.quality = options.quality;
+      modifyState.lossless = false;
+    } else {
+      modifyState.lossless = true;
+      modifyState.slightLoss = 100 - options.quality;
+    }
```

Comment:
```suggestion
    if (options.quality > 95) {
      modifyState.lossless = true;
      modifyState.slightLoss = 100 - options.quality;
    } else {
      modifyState.quality = options.quality;
      modifyState.lossless = false;
    }
```

Switched the branches because I wanted the comment to align with the condition.

Reply:
Good call. This happened because a comment in wp2 was incorrect. I've filed a bug for that https://bugs.chromium.org/p/webp2/issues/detail?id=3